### ISSUE TITLE
Improve command self-documentation

### ISSUE THREAD

**Issue by Maijin** [radare/radare2#4459]
Benefits of having a unique file for help:

1. No hackish code for every different help, one unique tree search
2. `[dD][ajbrfils] [sz] [arch] [bits]` this Usage parameter should be generated
3. Easy to contribute since doesn't require touching code
4. Unique error message when not existing self-doc : "Help doesn't exist, please contribute by adding to help.db"
5. File could be consumed by the HUD therefore automagically adding tons of feature to it
6. Complaining user will have all commands available stuff in only one file always updated
7. Can help for test since we'll have list of all commands in one file
8. Could be used for Localisation/Translation

**Comment by Ristovski** [comment]
A good question is, would all this be autogenerated? If so, what kind of system would it use, would it just grab strings by id from a file (so we can have localization)?
Otherwise seems like a great idea :smile:

**Comment by radare** [comment]
Im against adding localization to r2. It would be harder to maintain and will be slower

**Comment by Maijin** [comment]
Localisation is one between 10 benefits, can you comment on the whole idea ?

**Comment by XVilka** [comment]
To gain the power of radare2 wit,
The one should learn a little Brit.

++ for no localisation.

**Comment by sushant94** [comment]
@radare @XVilka @Maijin 
Ok, I had an idea for this a while back. Feel free to comment / review the same.

The basic idea was to build a compile time [trie](https://en.wikipedia.org/wiki/Trie). So adding a new command to radare2 shell would look something like this (disregard the names of functions/macros):

```C
// This is a macro that builds the trie from registered commands
r2_add_command ("pd", <pointer to function that handles pd>, <help for this command as a string>)
```

So, now when a user enters `p?`, it traverses the trie and prints help for all commands that `p` is a prefix of. This also solves another problem of sorting the commands manually. We can continue having the current separation of defining commands in a modular way.

If this idea seems good, I can elaborate more on this later.

**Comment by radare** [comment]
another idea was to use sdb, but sdb is plain k=v, there are no trees that can resolve sub-commands without dupping info.

**Comment by Maijin** [comment]
@MaskRay  there was also some attempts and discussions here you may want to check

**Comment by Maijin** [comment]
see https://github.com/radare/radare2/issues/5383

**Comment by karliss** [comment]
Related to  #16782

**Comment by ret2libc** [comment]
> Related to #16782

Yeah, probably we should indeed define some ".db" file or something where to store all the help messages and just load those from the file instead of hardcoding them in the code. Another random idea (not sure if it's good) could be to have a `.db` file that at compile time is converted into a `.c` file and just include that in the code, so it would become part of the binary anyway.
